### PR TITLE
Create two functions to extract canonical domain name and relative URL

### PR DESCRIPTION
**Pull request by Legai**
.idea/* was added to .gitignore in root directory

### CODE REVIEW

File: backend/app/middleware/canonizeURL.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const URL = require('url');
+const normalizeUrl = require('normalize-url');
+
+module.exports.getHost = (incomingURL) => {
+    return URL.parse(normalizeUrl(incomingURL)).host.toLowerCase();
+};
+module.exports.getPath = (incomingURL) => {
+    return URL.parse(normalizeUrl(incomingURL)).path.toLowerCase();
```
Comment (maxmalov) [inline backend/app/middleware/canonizeURL.js]: This looks the same as helper above. Consider extracting common logic into a separate function

File: backend/package.json [hunk 2]
```diff
@@ -7,6 +7,8 @@
   },
   "dependencies": {
     "express": "~4.14.1",
-    "pug": "~2.0.0-beta10"
+    "normalize-url": "^1.9.1",
+    "pug": "~2.0.0-beta10",
+    "url": "^0.11.0"
```
Comment (maxmalov) [inline backend/package.json]: You always should be common with standard modules https://nodejs.org/api/url.html

File: backend/app/middleware/canonizeURL.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const URL = require('url');
+const normalizeUrl = require('normalize-url');
+
+const normalizeAndParse = (url) => {
```
Comment (maxmalov) [inline backend/app/middleware/canonizeURL.js]: Consider using function declarations for top level functions. Arrow functions are helpful only as a callbacks

File: backend/app/middleware/canonizeURL.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const URL = require('url');
+const normalizeUrl = require('normalize-url');
+
+const normalizeAndParse = function(url){
```
Comment (102) [inline backend/app/middleware/canonizeURL.js]: `function normalizeAndParse(url)`

File: .gitignore [hunk 5]
```diff
@@ -35,3 +35,5 @@ jspm_packages
 
 # Optional REPL history
 .node_repl_history
+
+.idea/*
```
Comment (102) [inline .gitignore]: `.idea/`

**Comment by 102** [comment]
You also have to make sure that or title has no typos since it will be shown in repo history after GitHub's squash merge

File: backend/app/middleware/canonizeURL.js (hunk 4 above)
Comment (maxmalov) [inline backend/app/middleware/canonizeURL.js]: :+1:

File: .gitignore (hunk 5 above)
Comment (maxmalov) [inline .gitignore]: And even `/.idea

Comment (IgorZiegler) [inline .gitignore]: use gitexcludes

File: backend/app/middleware/canonizeURL.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const URL = require('url');
```
Comment (maxmalov) [inline backend/app/middleware/canonizeURL.js]: One more thing, this module isn't a middleware. You should consider find an appropriate term for this kind of logic.

**Pull request merged**